public/views/index.html

```diff
@@ -14,6 +14,15 @@
 
     <base href="[[.AppSubUrl]]/" />
 
+    <script>
+      // NI fork: Ensure that grafana is loaded within an iframe unless it is running locally
+      if (window.self === window.top && window.location.hostname !== 'localhost') {
```

Comment:
I'm wondering if this becomes a whack-a-mole. There are a bunch of hostnames for localhost, i.e. the common `127.0.0.1` but really `127.*.*.*`, `0.0.0.0`, the ip-v6 `::1`, `*.localhost`, `localhost.*`, `*.localhost.*`, but also the common convention of `*.local`.

Only other convention I can think of is instead having a hash in the url like `#skip-redirect` we check for. Something like that could also be useful for integration testing maybe?

Reply:
I wasn't thrilled about this check either for the exact reason you mentioned of having so many ways to say "localhost" in a URL. I ultimately convinced myself that it was acceptable because the docs point to grafana running at `http://localhost:3000`, and that if we ran into more scenarios that needed to be handled, we could reevaluate.

That being said, I don't mind going with a different approach. I tried the hash approach, but I think it'll ultimately be pretty annoying without additional changes. The issue is that as you navigate around grafana, the hash is lost. Therefore, if you refresh the page after using grafana for a little bit, the hash doesn't match what we're looking for, which ultimately results in grafana continually reloading.